src/python/aqueduct_executor/operators/function_executor/execute.py

```diff
@@ -166,3 +171,29 @@ def run(spec: FunctionSpec) -> None:
         print(f"Failed with system error. Full Logs:\n{exec_state.json()}")
         utils.write_exec_state(storage, spec.metadata_path, exec_state)
         sys.exit(1)
+
+
+def run_with_setup(spec: FunctionSpec) -> None:
+    """
+    Performs the setup needed for a Function operator and then executes it.
+    """
+    # Generate a unique function extract path if one does not exist already
+    fn_extract_path = None
+    if not spec.function_extract_path:
+        fn_extract_path = os.path.join(os.getcwd(), str(uuid.uuid4()))
+        spec.function_extract_path = fn_extract_path
+
+    op_path = get_extract_path.run(spec)
+
+    extract_function.run(spec)
+
+    requirements_path = os.path.join(op_path, "requirements.txt")
+    if os.path.exists(requirements_path):
+        os.system("pip3 install -r {}".format(requirements_path))
+    
+    run(spec)
+
+    if fn_extract_path:
```

Comment:
TODO: Figure out how to perform cleanup if there is an error before this point

Reply:
https://linear.app/aqueducthq/issue/ENG-1496/clean-up-extracted-function-files-in-airflow-execution